modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/ScanNode.java

```diff
@@ -96,13 +96,16 @@ private void doPush() throws Exception {
 
     /** */
     private void push() throws Exception {
+        if (isClosed())
```

Comment:
this is should be at the very beginning of the task execution, but for ScanNode the task scheduled to the executor is `doPush`, and there is call to `checkstate()` causing an exception. Let's get rid of `doPush` at all and place a `checkstate()` right after an `isClosed()` validation.